perception/autoware_tensorrt_bevformer/include/autoware/tensorrt_bevformer/bevformer_node.hpp

```diff
@@ -0,0 +1,211 @@
+#include "autoware/tensorrt_bevformer/ros_utils.hpp"
+#include "tf2_ros/buffer.h"
+#include "tf2_ros/transform_listener.h"
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <rclcpp/logging.hpp>
+
+#include "autoware_localization_msgs/msg/kinematic_state.hpp"
+#include "visualization_msgs/msg/marker_array.hpp"
+#include <geometry_msgs/msg/quaternion.hpp>
+#include <geometry_msgs/msg/vector3.hpp>
+#include <sensor_msgs/msg/image.hpp>
+
+#include <cv_bridge/cv_bridge.h>
+#include <message_filters/subscriber.h>
+#include <message_filters/sync_policies/approximate_time.h>
+#include <message_filters/synchronizer.h>
+
+#include <map>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+namespace autoware
+{
+namespace tensorrt_bevformer
+{
+class TRTBEVFormerNode : public rclcpp::Node
+{
+private:
+  size_t img_N_;
+  uint32_t src_img_w_;
+  uint32_t src_img_h_;
+  std::map<std::string, int> model_shape_params_;
+  std::map<std::string, std::vector<std::string>> model_output_shapes_;
+
+  std::string engine_file_;
+  std::string plugin_path_;
+
+  std::vector<std::string> class_names_;
+
+  std::vector<double> pc_range_;
+  std::vector<double> post_center_range_;
+
+  std::string onnx_file_;
+  int workspace_size_;
+  bool auto_convert_;
+  std::string precision_;
+
+  std::map<std::string, std::vector<double>> reshapeTensorOutputs(
+    const std::map<std::string, std::vector<double>> & rawOutputs);
+
+  // Static transforms (calculated once)
+  Eigen::Quaterniond lidar2ego_rot_static_;
+  Eigen::Translation3d lidar2ego_trans_static_;
+  bool lidar2ego_transforms_ready_ = false;
+
+  // Inference components
+  float score_thre_;         // Score threshold for object detection
+  bool has_twist_ = true;    // whether set twist for objects
+  bool debug_mode_ = false;  // Flag to enable debug mode for nuscenes marker visualization
+
+  // Publishers and subscribers
+  rclcpp::Publisher<autoware_perception_msgs::msg::DetectedObjects>::SharedPtr pub_boxes_;
+  rclcpp::Publisher<visualization_msgs::msg::MarkerArray>::SharedPtr pub_markers_;
+
+  // Camera subscriptions
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr sub_f_caminfo_;
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr sub_fl_caminfo_;
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr sub_fr_caminfo_;
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr sub_b_caminfo_;
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr sub_bl_caminfo_;
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr sub_br_caminfo_;
+
+  // Initialization flags
+  std::vector<bool>
+    caminfo_received_;  // Flag indicating if camera info has been received for each camera
+  bool camera_info_received_flag_ = false;
+
+  // can_bus subscription
+  message_filters::Subscriber<autoware_localization_msgs::msg::KinematicState> sub_can_bus_;

```

Comment:
I suggest to update the subscriber name to match with what it is subscribing. 

```suggestion
  message_filters::Subscriber<autoware_localization_msgs::msg::KinematicState> sub_kinematic_state;

```

Reply:
Thanks for pointing that out. We'll update the subscriber name to match the subscribed topic (sub_kinematic_state).